frame/evm/precompile/bn128/src/lib.rs

```diff
@@ -282,8 +282,7 @@ impl Precompile for Bn128Pairing {
 			}
 		};
 
-		let mut buf = [0u8; 32];
-		ret_val.to_big_endian(&mut buf);
+		let buf = ret_val.to_big_endian();
```

Comment:
What about updating LINE 290 instead of assigning a new buf here?

```patch
		Ok(PrecompileOutput {
			exit_status: ExitSucceed::Returned,
-			output: buf.to_vec(),
+			output: ret_val.to_big_endian().to_vec(),
		})
```